api/tests/provider_api_tests.py

```diff
@@ -13,42 +13,63 @@ class BaseAPITest(APITestCase):
     def setUp(self):
         self.request_factory = RequestFactory()
         self.view_set = ProviderListViewSet
-        self.base_url = '/api/providers/'
 
-    def get_request(self, id=None):
-        request = self.request_factory.get(f'{self.base_url}{id}')
+    def get_request_factory(self, url, average_ratings={}, id=None):
+        request = self.request_factory.get(url, average_ratings)
         return request
 
-    def get_retrieve_response(self, id=None):
-        request = self.get_request(id)
+    def get_request_client(self, url, average_ratings={}):
+        request = self.client.get(url, average_ratings)
```

Comment:
There are some methods that are duplicated meaning that in some we create ourselves the request and response and in this one we use the client. Try to use only the `self.client` as it does already handle most of the util functions you have here.